allensdk/brain_observatory/behavior/behavior_project_cache/tables/metadata_parsers.py

```diff
@@ -0,0 +1,249 @@
+import pandas as pd
+
+#####################################################################
+#
+#                   VBO PROJECT CONSTANTS
+#
+#####################################################################
+
+
+SESSION_TYPES = [
+    "OPHYS_0_images_A_habituation",
+    "OPHYS_0_images_B_habituation",
+    "OPHYS_0_images_G_habituation",
+    "OPHYS_1_images_A",
+    "OPHYS_1_images_B",
+    "OPHYS_1_images_G",
+    "OPHYS_2_images_A_passive",
+    "OPHYS_2_images_B_passive",
+    "OPHYS_2_images_G_passive",
+    "OPHYS_3_images_A",
+    "OPHYS_3_images_B",
+    "OPHYS_3_images_G",
+    "OPHYS_4_images_A",
+    "OPHYS_4_images_B",
+    "OPHYS_4_images_H",
+    "OPHYS_5_images_A_passive",
+    "OPHYS_5_images_B_passive",
+    "OPHYS_5_images_H_passive",
+    "OPHYS_6_images_A",
+    "OPHYS_6_images_B",
+    "OPHYS_6_images_H",
+    "TRAINING_0_gratings_autorewards_15min",
+    "TRAINING_1_gratings",
+    "TRAINING_2_gratings_flashed",
+    "TRAINING_3_images_A_10uL_reward",
+    "TRAINING_3_images_B_10uL_reward",
+    "TRAINING_3_images_G_10uL_reward",
+    "TRAINING_4_images_A_handoff_lapsed",
+    "TRAINING_4_images_A_handoff_ready",
+    "TRAINING_4_images_A_training",
+    "TRAINING_4_images_B_training",
+    "TRAINING_4_images_G_training",
+    "TRAINING_5_images_A_epilogue",
+    "TRAINING_5_images_A_handoff_lapsed",
+    "TRAINING_5_images_A_handoff_ready",
+    "TRAINING_5_images_B_epilogue",
+    "TRAINING_5_images_B_handoff_lapsed",
+    "TRAINING_5_images_B_handoff_ready",
+    "TRAINING_5_images_G_epilogue",
+    "TRAINING_5_images_G_handoff_lapsed",
+    "TRAINING_5_images_G_handoff_ready",
+]
+
+PROJECT_CODES = [
+    "VisualBehavior",
+    "VisualBehaviorTask1B",
+    "VisualBehaviorMultiscope",
+    "VisualBehaviorMultiscope4areasx2d",
+]
+
+NUM_STRUCTURES_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 2,
+    "VisualBehaviorMultiscope4areasx2d": 4,
+}
+
+NUM_DEPTHS_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 4,
+    "VisualBehaviorMultiscope4areasx2d": 2,
+}
+
+OPHYS_BEHAV_TYPE_DICT = {
+    "active_behavior": ["0", "1", "3", "4", "6"],
+    "passive_viewing": ["2", "5", "7"],
+}
+
+
+############################
+#
+#     from session type
+#
+############################
+
+
+def parse_behavior_context(metadata_row: pd.Series) -> str:
+    """Return if session is passive or active.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    behavior_type : str
+        Return string describing active or passive session.
+    """
+    passive_ophys_session_nums = ["2", "5", "7"]
+
+    session_type = metadata_row["session_type"]
+
+    if session_type not in SESSION_TYPES:
+        return None
+    is_passive = (
+        any(
+            passive_num in session_type
```

Comment:
If the code crashes because `session_type` is some other type that would be desirable. Non-string `session_type` likely means something else is going wrong.